msteams-platform/concepts/bots/bots-files.md

```diff
@@ -0,0 +1,152 @@
+
+## Receiving files in 1:1 chat
+
+When a user sends a file to your bot, the file is first uploaded to the user's OneDrive for Business storage. Your bot will then receive a message activity notifying you of the user upload. The activity will contain file metadata, such as its name and the content URL, and the content URL's expiry time (e.g. 24 hours). You can directly read from this URL to fetch its binary content.
+
+### Message activity with file attachment example
+
+```json
+{
+  "attachments": [{
+    "contentType": "application/vnd.microsoft.teams.file.download.info",
+    "contentUrl": "https://contoso.sharepoint.com/personal/johnadams_contoso_com/Documents/Applications/file_example.txt",
+    "name": "file_example.txt",
+    "content": {
+      "downloadUrl" : "https://download.link",
+      "uniqueId": "1150D938-8870-4044-9F2C-5BBDEBA70C9D",
+      "fileType": "txt",
+      "etag": "123"
+    }
+  }]
+}
+```
+
+As a best practice, you should acknowledge the file upload by sending back a message to the user.
+
+## Uploading files to 1:1 chat
+
+Uploading a file to a user involves the following steps:
+
+1. Send a message to the user requesting permission to write the file. This message must contain a `FileConsentCard` attachment with the name of the file to be uploaded.
+2. If the user accepts the file download, your bot will receive an Invoke activity with a location URL.
+3. To transfer the file, your bot performs an HTTP POST directly into the provided location URL.
+4. Optionally, you can perform a message update on the original message, removing the consent card if you do not want to allow the user to accept further uploads of the same file.
+
+### Message requesting permission to upload
+
+This message contains a simple attachment object requesting user permission to upload the file.
+
+```json
+{
+  "attachments": [{
+    "contentType": "application/vnd.microsoft.teams.card.file.consent",
+    "name": "file_example.txt",
+    "content": {
+      "description": "<Purpose of the file, e.g. this is your monthly expense report>",
+      "sizeInBytes": 1029393,
+      "acceptContext": {
+      },
+      "declineContext": {
+      }
+    }
+  }]
+}
+```
+
+### Invoke activity when the user accepts the file
+
+An activity is sent to your bot if and when the user accepts the file. It contains the OneDrive for Business placeholder URL that your bot can then Post into to transfer the file contents. The following example shows the attachment object that your bot will receive:
+
+```json
+{
+  "value": {
+    "type": "fileUpload",
+    "action": "accept",
+    "context": {
+    },
+    "uploadInfo": {
+      "contentUrl": "https://contoso.sharepoint.com/personal/johnadams_contoso_com/Documents/Applications/file_example.txt",
+      "name": "file_example.txt",
+      "uploadUrl": "https://upload.link",
+      "uniqueId": "1150D938-8870-4044-9F2C-5BBDEBA70C8C",
+      "fileType": "txt",
+      "etag": "123"
```

Comment:
I am not getting this etag in my bot. Can this file even have an etag, when the file doesn't have any content yet?